### PR TITLE
Add time import stamp field in metadata response

### LINKED ISSUE #122: Expose the "ImportTimeStamp" field from VolumeDataLayout through /metadata

### PR DESCRIPTION
**Pull request by AdilKhurshid**
closes #122

### CODE REVIEW

**Comment by achaikou** [comment]
Haven't looked at it yet, but there is one general problem with this task.
Every time we would need to regenerate our test files we would need to update that timestamp simply because it is regenerated.
(we do happen to update test files once in a while when we need to import some feature we did not have there before).

Any ideas? I am not sure today if the best approach is
- to just accept having to deal with that once in a while and frustrate every developer who would have to investigate this test that fails out of the blue
- try to check for date regex instead of a defined date
- have a separate file just for date and check date only in its own test
- something else

**Comment by AdilKhurshid** [comment]
> Haven't looked at it yet, but there is one general problem with this task. Every time we would need to regenerate our test files we would need to update that timestamp simply because it is regenerated. (we do happen to update test files once in a while when we need to import some feature we did not have there before).
> 
> Any ideas? I am not sure today if the best approach is
> 
> * to just accept having to deal with that once in a while and frustrate every developer who would have to investigate this test that fails out of the blue
> * try to check for date regex instead of a defined date
> * have a separate file just for date and check date only in its own test
> * something else

Yeah, I agree it should be something generic, I will investigate the regex approach to fix this. Thanks 😊

File: internal/core/core_test.go [hunk 1]
```diff
@@ -88,7 +90,15 @@ func TestMetadata(t *testing.T) {
...
-	require.Equal(t, meta, expected)
+	require.Equal(t, meta.Axis, expected.Axis)
+	require.Equal(t, meta.BoundingBox, expected.BoundingBox)
+	require.Equal(t, meta.Crs, expected.Crs)
+	require.Equal(t, meta.InputFileName, expected.InputFileName)
+	require.Equal(
+		t,
+		true,
+		regexp.MustCompile(expected.ImportTimeStamp).MatchString(meta.ImportTimeStamp),
+		)
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply internal/core/core_test.go:101]: agree 👍

File: cmd/query/main_test.go [hunk 2]
```diff
@@ -354,11 +355,30 @@ func TestMetadataHappyHTTPResponse(t *testing.T) {
...
-		require.JSONEqf(t, expectedMetadata, metadata, "Metadata not equal in case '%s'", testcase.name)
+		var expectedMap map[string]any
+		var actualMap map[string]any
+
+		json.Unmarshal([]byte(expectedMetadata), &expectedMap)
+		json.Unmarshal([]byte(metadata), &actualMap)
+
+		val, ok := actualMap["importTimeStamp"]
+		if !ok {
+			t.Errorf("Metadata not equal in case '%s'", testcase.name)
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply cmd/query/main_test.go:371]: sure

File: cmd/query/main_test.go [hunk 3]
```diff
@@ -354,11 +355,30 @@ func TestMetadataHappyHTTPResponse(t *testing.T) {
...
+		}
+		actualVal   := val.(string)
+		expectedVal := expectedMap["importTimeStamp"].(string)
+		if !regexp.MustCompile(expectedVal).MatchString(actualVal) {
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply cmd/query/main_test.go:375]: sure

File: tests/e2e/test_requests.py [hunk 4]
```diff
@@ -132,24 +133,32 @@ def test_fence(method):
...
-        "inputFileName": "well_known.segy"
+        "crs"            : "utmXX",
+        "inputFileName"  : "well_known.segy",
+        "importTimeStamp": "^\\d{4}-\\d{2}-\\d{2}[A-Z]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}[A-Z]$"
     }
-    """)
-    assert metadata == expected_metadata
+
+    expected_val = expected_metadata.get("importTimeStamp")
+    actual_val   = metadata.get("importTimeStamp")
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply tests/e2e/test_requests.py:154]: sure

[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply tests/e2e/test_requests.py:154]: sure

File: cmd/query/main_test.go [hunk 5]
```diff
@@ -366,19 +365,16 @@ func TestMetadataHappyHTTPResponse(t *testing.T) {
...
-			actualMap["importTimeStamp"]   = "dummy"
-		}
-		require.Equal(t, expectedMap, actualMap, "Metadata not equal in case '%s'", testcase.name)
+
+		require.Regexp(t, expectedMap["importTimeStamp"], actualMap["importTimeStamp"])
+
+		expectedMap["importTimeStamp"] = "dummy"
+		actualMap["importTimeStamp"]   = "dummy"
+
+		require.Equal(t, expectedMap, actualMap, "importTimeStampt is not found in case '%s'", testcase.name)
```
Comment (achaikou) [inline cmd/query/main_test.go:377]: And here error message "Metatadata not equal..." was more correct :)

**Review by achaikou — COMMENTED** [review]
After error message fix we are ready for merge!

File: cmd/query/main_test.go (hunk 5 above)
Comment (AdilKhurshid) [reply cmd/query/main_test.go]: done :) https://github.com/equinor/vds-slice/pull/172/files#diff-079081f30f9137f135638d9b5e2a772d22a7232c750ab2dc087141318c8e168fR377

**Review by achaikou — APPROVED** [review]

**Pull request merged**